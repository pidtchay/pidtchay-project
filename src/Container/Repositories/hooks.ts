/* eslint-disable jsdoc/require-returns */
import { useDispatch } from 'react-redux';
import { getRepositoriesData } from 'Store/repositories/api';
import { listRepositoriesSlice } from 'Store/repositories/slice';

/**
 * UseGithubRepos.
 */
export function useGithubRepos() {
    const dispatch = useDispatch();

    const { clean, fetched, fetching, fetchingError } = listRepositoriesSlice.actions;

    const fetchData = async (orgName: string) => {
        dispatch(fetching());
        return getRepositoriesData(orgName)
            .then((githubRepos) => {
                console.debug({ githubRepos });
                dispatch(fetched({ repositories: githubRepos }));
            })
            .catch((error) => {
                error.clientMessage = "Can't get repositories data";
                dispatch(fetchingError({ error }));
            });
    };

    const cleanData = () => {
        dispatch(clean());
    };

    return { fetchData, cleanData };
}
