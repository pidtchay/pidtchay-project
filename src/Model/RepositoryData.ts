/**
 * Data model for query result on github api.
 *
 * @property {string} [avatarImg] Avatag image.
 * @property {string} [description] Repository description.
 * @property {string} [forks] Repository forks.
 * @property {string} [homepage] Repository homepage.
 * @property {number} [id] Id.
 * @property {string} [language] Used programming language.
 * @property {string} [name] Repository name.
 * @property {string} [owner] Repository owner.
 * @property {string} [type] Type.
 * @property {string} [watchers] Repository watchers.
 */
export interface IRepositoryData {
    avatarImg: string;
    description: string;
    forks: number;
    homepage: string;
    id: number;
    language: string;
    name: string;
    owner: string;
    type: string;
    watchers: number;
}
