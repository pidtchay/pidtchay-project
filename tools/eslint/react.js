module.exports = {
    'react/no-did-update-set-state': 0,
    'react-hooks/exhaustive-deps': 0,
    'react/state-in-constructor': [2, 'never'],
    'react/destructuring-assignment': 0,
    'react/no-danger': 2,
    'react/prefer-stateless-function': 1,
    'react/no-array-index-key': 1,
    'react/jsx-props-no-spreading': 0,
    'react/static-property-placement': ['warn', 'property assignment', { // Указывает, где должны быть размещены статические свойства компонента React
      'childContextTypes': 'static getter',
      'contextTypes': 'static public field',
      'contextType': 'static public field',
      'displayName': 'static public field',
    }],
    'react/require-default-props': 1,
    'react/sort-comp': [1, { order: ['static-methods', 'lifecycle', 'everything-else', 'render']}],

}