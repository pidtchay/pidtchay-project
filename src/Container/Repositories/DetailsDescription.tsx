import { Descriptions, Tag } from 'antd';
import * as React from 'react';
import { useLiteralValue } from 'Utils/hooks';

interface IDetailsDescriptionProps {
    name: string;
    description: string;
    language: string;
}

export const DetailsDescription: React.FC<IDetailsDescriptionProps> = ({
    name,
    description,
    language
}) => {
    const { getValue: getLiteralValue } = useLiteralValue();
    return (
        <>
            <Descriptions
                title={getLiteralValue(
                    'Pages.GithubRepositories.Details.description'
                )}
            >
                <Descriptions.Item
                    label={getLiteralValue(
                        'Pages.GithubRepositories.Details.name'
                    )}
                >
                    {name}
                </Descriptions.Item>
                <Descriptions.Item
                    label={getLiteralValue(
                        'Pages.GithubRepositories.Details.description'
                    )}
                >
                    {description}
                </Descriptions.Item>
                <Descriptions.Item
                    label={getLiteralValue(
                        'Pages.GithubRepositories.Details.language'
                    )}
                >
                    <Tag color="geekblue">{language}</Tag>
                </Descriptions.Item>
            </Descriptions>
        </>
    );
};
