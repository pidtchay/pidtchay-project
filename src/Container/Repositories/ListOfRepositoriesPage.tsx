import * as React from 'react';
import { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { RootState } from 'Store/root';
import { convertRepositoryDataToSchema, convertRepositoryDataToColumns } from 'Utils/TableUtils';
import { ListOfRepositories } from './ListOfRepositories';
import { useGithubRepos } from './hooks';

export const ListOfRepositoriesPage = () => {
    const gitgubRepos = useSelector((state: RootState) => state.githubRepos);
    const [organisationName, setOrganosationName] = useState('facebook');
    const { fetchData } = useGithubRepos();

    useEffect(() => {
        fetchData(organisationName);
    }, [organisationName]);

    const handleFetchRepositories = (value: string) => {
        setOrganosationName(value);
    };

    return <ListOfRepositories columns={convertRepositoryDataToColumns(convertRepositoryDataToSchema(gitgubRepos.repositories[0]))} data={gitgubRepos.repositories} onFetchRepositories={handleFetchRepositories} />;
};
