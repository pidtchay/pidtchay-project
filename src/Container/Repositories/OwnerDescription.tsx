import { Descriptions, Button } from 'antd';
import * as React from 'react';
import { useLiteralValue } from 'Utils/hooks';

interface IOwnerDescriptionProps {
    owner: string;
    homepage: string;
    forks: number;
    watchers: number;
}

export const OwnerDescription: React.FC<IOwnerDescriptionProps> = ({
    owner,
    homepage,
    forks,
    watchers
}) => {
    const { getValue: getLiteralValue } = useLiteralValue();
    return (
        <>
            <Descriptions
                title={getLiteralValue('Pages.GithubRepositories.Details.info')}
            >
                <Descriptions.Item
                    label={getLiteralValue(
                        'Pages.GithubRepositories.Details.owner'
                    )}
                >
                    {owner}
                </Descriptions.Item>
                <Descriptions.Item
                    label={getLiteralValue(
                        'Pages.GithubRepositories.Details.home_page'
                    )}
                >
                    <Button type="link">{homepage}</Button>
                </Descriptions.Item>
                <Descriptions.Item
                    label={getLiteralValue(
                        'Pages.GithubRepositories.Details.forks'
                    )}
                >
                    {forks}
                </Descriptions.Item>
                <Descriptions.Item
                    label={getLiteralValue(
                        'Pages.GithubRepositories.Details.watchers'
                    )}
                >
                    {watchers}
                </Descriptions.Item>
            </Descriptions>
        </>
    );
};
