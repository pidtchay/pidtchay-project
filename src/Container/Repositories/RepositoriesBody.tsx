/* eslint-disable jsdoc/require-returns */
/* eslint-disable jsdoc/require-param */
import { Input, BackTop, Tooltip } from 'antd';
import * as React from 'react';
import { RocketOutlined } from '@ant-design/icons';
import { TableComponent } from 'Component/Common/TableComponent';
import { IRepositoryData } from 'Model/RepositoryData';
import { IColumnField } from 'Model/Table';
import style from 'Style/RepositoriesBody.less';
import { convertStringArrayToString } from 'Utils/common';
import { useLiteralValue } from 'Utils/hooks';

const { Search } = Input;

/**
 * @property {ILiterals} [literals] Language settings in the form of key-value. The text depends on the selected language.
 * @property {IColumnField[]} [columns] Table columns headers that will be visible to the user.
 * @property {IRepositoryData[]} [data] List of repositories for the selected organization name.
 * @property {Function} [onFetchRepositories] The function queries the list of repositories for the selected organization name.
 */
interface IRepositoriesBodyProps {
    columns: IColumnField[];
    data: IRepositoryData[];
    onFetchRepositories: (valuse: string) => void;
}

/**
 * Repositories Body Component.
 */
export const RepositoriesBody: React.FC<IRepositoriesBodyProps> = ({
    columns,
    data,
    onFetchRepositories
}) => {
    const { getValue: getLiteralValue } = useLiteralValue();
    const placeholder = convertStringArrayToString(
        getLiteralValue('Pages.GithubRepositories.search_placeholder')
    );
    return (
        <>
            <div>
                <Search
                    placeholder={placeholder}
                    enterButton="Search"
                    size="large"
                    onSearch={onFetchRepositories}
                />
            </div>
            <TableComponent columns={columns} dataSource={data || []} />
            <Tooltip title={getLiteralValue('ACTIONS.up')}>
                <BackTop className={style.repositories_body}>
                    <div className={style.repositories_body_up_button}>
                        <RocketOutlined />
                    </div>
                </BackTop>
            </Tooltip>
        </>
    );
};
