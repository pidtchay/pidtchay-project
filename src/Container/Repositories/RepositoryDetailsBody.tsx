import {
    Divider,
    Row,
    Col,
    Avatar,
    Descriptions,
    Tag,
    Button,
    PageHeader
} from 'antd';
import * as React from 'react';
import { ILiterals } from 'Model/Literals';
import { IRepositoryData } from 'Model/RepositoryData';
import style from 'Style/RepositoryDetails.less';
import { getI18nValue } from 'Utils/common';

interface IProps {
    literals: ILiterals;
    repository: IRepositoryData;
    onBackClick: () => void;
}

export const RepositoryDetails: React.FC<IProps> = ({
    literals,
    repository,
    onBackClick: handleBackClick
}) => {
    console.debug({ literals });
    return (
        <div className={style.details_layout}>
            <PageHeader
                onBack={handleBackClick}
                title={
                    getI18nValue(literals, 'Pages.GithubRepositories.back') ||
                    ''
                }
            />
            <Divider orientation="left">
                {getI18nValue(
                    literals,
                    'Pages.GithubRepositories.Details.Divider.owner'
                )}
            </Divider>
            <Row gutter={[16, 16]}>
                <Col span={4}>
                    <div className={style.avatar_panel}>
                        <Avatar
                            shape="square"
                            size={100}
                            src={repository.avatarImg}
                        />
                    </div>
                </Col>
                <Col span={10} offset={2}>
                    <Descriptions
                        title={getI18nValue(
                            literals,
                            'Pages.GithubRepositories.Details.info'
                        )}
                    >
                        <Descriptions.Item
                            label={getI18nValue(
                                literals,
                                'Pages.GithubRepositories.Details.owner'
                            )}
                        >
                            {repository.owner}
                        </Descriptions.Item>
                        <Descriptions.Item
                            label={getI18nValue(
                                literals,
                                'Pages.GithubRepositories.Details.home_page'
                            )}
                        >
                            <Button type="link">{repository.homepage}</Button>
                        </Descriptions.Item>
                        <Descriptions.Item
                            label={getI18nValue(
                                literals,
                                'Pages.GithubRepositories.Details.forks'
                            )}
                        >
                            {repository.forks}
                        </Descriptions.Item>
                        <Descriptions.Item
                            label={getI18nValue(
                                literals,
                                'Pages.GithubRepositories.Details.watchers'
                            )}
                        >
                            {repository.watchers}
                        </Descriptions.Item>
                    </Descriptions>
                </Col>
            </Row>
            <Divider orientation="left">
                {getI18nValue(
                    literals,
                    'Pages.GithubRepositories.Divider.details'
                )}
            </Divider>
            <Row gutter={[16, 16]}>
                <Col>
                    <Descriptions
                        title={getI18nValue(
                            literals,
                            'Pages.GithubRepositories.Details.description'
                        )}
                    >
                        <Descriptions.Item
                            label={getI18nValue(
                                literals,
                                'Pages.GithubRepositories.Details.name'
                            )}
                        >
                            {repository.name}
                        </Descriptions.Item>
                        <Descriptions.Item
                            label={getI18nValue(
                                literals,
                                'Pages.GithubRepositories.Details.description'
                            )}
                        >
                            {repository.description}
                        </Descriptions.Item>
                        <Descriptions.Item
                            label={getI18nValue(
                                literals,
                                'Pages.GithubRepositories.Details.language'
                            )}
                        >
                            <Tag color="geekblue">{repository.language}</Tag>
                        </Descriptions.Item>
                    </Descriptions>
                </Col>
            </Row>
        </div>
    );
};
