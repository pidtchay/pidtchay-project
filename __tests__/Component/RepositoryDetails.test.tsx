import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { RepositoryDetails } from 'Container/Repositories/RepositoryDetailsBody';
import { IRepositoryData } from 'Model/RepositoryData';

describe('render success',() => {
    it('should match the snapshot', () => {
        const handleBackClick = jest.fn();
        const repository: IRepositoryData = {
            avatarImg: 'http://test.com/avatar.jpeg',
            description: '',
            forks: 1,
            homepage: 'http://test.com/home',
            id: 1,
            language: 'javascript',
            name: 'ReactJS Todo list',
            owner: 'Monster corp',
            type: '',
            watchers: 5
        };
        const component = renderer.create(<RepositoryDetails onBackClick={handleBackClick} repository={repository} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
