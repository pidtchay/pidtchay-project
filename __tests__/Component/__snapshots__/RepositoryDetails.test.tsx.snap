// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render success should match the snapshot 1`] = `
<div>
  <div
    className="ant-page-header ant-page-header-ghost"
  >
    <div
      className="ant-page-header-heading"
    >
      <div
        className="ant-page-header-heading-left"
      >
        <div
          className="ant-page-header-back"
        >
          <div
            aria-label="Back"
            className="ant-page-header-back-button"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            role="button"
            style={
              Object {
                "background": "transparent",
                "border": 0,
                "display": "inline-block",
                "lineHeight": "inherit",
                "padding": 0,
              }
            }
            tabIndex={0}
          >
            <span
              aria-label="arrow-left"
              className="anticon anticon-arrow-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="arrow-left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 000 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </span>
          </div>
        </div>
        <span
          className="ant-page-header-heading-title"
          title="Back to repositories"
        >
          Back to repositories
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-left"
    role="separator"
  >
    <span
      className="ant-divider-inner-text"
    >
      Owner
    </span>
  </div>
  <div
    className="ant-row"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-4"
      style={Object {}}
    >
      <div>
        <span
          className="ant-avatar ant-avatar-square ant-avatar-image"
          style={
            Object {
              "fontSize": 18,
              "height": 100,
              "lineHeight": "100px",
              "width": 100,
            }
          }
        >
          <img
            onError={[Function]}
            src="http://test.com/avatar.jpeg"
          />
        </span>
      </div>
    </div>
    <div
      className="ant-col ant-col-10 ant-col-offset-2"
      style={Object {}}
    >
      <div
        className="ant-descriptions"
      >
        <div
          className="ant-descriptions-view"
        >
          <table>
            <tbody>
              <tr
                className="ant-descriptions-row"
              >
                <td
                  className="ant-descriptions-item"
                  colSpan={1}
                >
                  <span
                    className="ant-descriptions-item-content"
                  >
                    Monster corp
                  </span>
                </td>
                <td
                  className="ant-descriptions-item"
                  colSpan={1}
                >
                  <span
                    className="ant-descriptions-item-content"
                  >
                    <button
                      className="ant-btn ant-btn-link"
                      onClick={[Function]}
                      type="button"
                    >
                      <span>
                        http://test.com/home
                      </span>
                    </button>
                  </span>
                </td>
                <td
                  className="ant-descriptions-item"
                  colSpan={1}
                >
                  <span
                    className="ant-descriptions-item-content"
                  >
                    1
                  </span>
                </td>
              </tr>
              <tr
                className="ant-descriptions-row"
              >
                <td
                  className="ant-descriptions-item"
                  colSpan={3}
                >
                  <span
                    className="ant-descriptions-item-content"
                  >
                    5
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-divider ant-divider-horizontal"
    role="separator"
  />
  <div
    className="ant-row"
    style={Object {}}
  >
    <div
      className="ant-col"
      style={Object {}}
    >
      <div
        className="ant-descriptions"
      >
        <div
          className="ant-descriptions-view"
        >
          <table>
            <tbody>
              <tr
                className="ant-descriptions-row"
              >
                <td
                  className="ant-descriptions-item"
                  colSpan={1}
                >
                  <span
                    className="ant-descriptions-item-content"
                  >
                    ReactJS Todo list
                  </span>
                </td>
                <td
                  className="ant-descriptions-item"
                  colSpan={1}
                >
                  
                </td>
                <td
                  className="ant-descriptions-item"
                  colSpan={1}
                >
                  <span
                    className="ant-descriptions-item-content"
                  >
                    <span
                      className="ant-tag ant-tag-geekblue"
                      style={
                        Object {
                          "backgroundColor": undefined,
                        }
                      }
                    >
                      javascript
                    </span>
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
